### PR TITLE
Restore checkpoint automatically based on some rules

### LINKED ISSUE #250: Checkpoint files should be saved in checkpoint_dir named after the commit hash
Instead of saving checkpoints in the directory

`checkpoint_dir`

we should create a sub directory of `checkpoint_dir` named after the current commit's hash `<commit_hash>`, then save the checkpoints under

`checkpoint_dir/<commit_hash>`

This would allow us sync the checkpoints with the apropos commit, thus allowing us to load old checkpoints even through the current operator graph is no longer compatible with the older checkpoints.

### PR DESCRIPTION
**Pull request by lissyx**
Fixes #250

### CODE REVIEW

**Comment by lissyx** [comment]
Should be ready to review @kdavis-mozilla

File: util/automation.py [hunk 1]
```diff
@@ -33,11 +33,20 @@
 DEEPSPEECH_CLONE_PATH  = os.path.abspath(os.environ.get('ds_clone_path', './ds_exec_clone/'))
 
 CACHE_DIR = os.path.join(BaseDirectory.xdg_cache_home, 'deepspeech_wer')
+DATA_DIR  = os.path.join(BaseDirectory.xdg_data_home, 'deepspeech_wer')
+
+# Lock file to prevent execution: avoid multiple simultaneous execution, blocks
+# execution if needed
 LOCKFILE  = os.path.join(CACHE_DIR, 'lock')
+
+# File holding the last SHA1 we processes
```
Comment (Cwiiis) [inline util/automation.py]: s/processes/processed/

File: util/automation.py [hunk 2]
```diff
@@ -423,30 +423,73 @@ def ensure_gpu_usage(root_dir):
...
+        os.unlink(CKPTFILE)
+
+        # Make sure the targetted path actually exists.
+        # We check here to not change the value of checkpoint_dir in case
+        # it does not exists.
+        # If it actually exists, the later os.path.isdir(checkpoint_dir) will
+        # take care of setting the checkpoint_restore value
+        print "Trying with maybe_checkpoint_dir", maybe_checkpoint_dir
+        if not os.path.isdir(maybe_checkpoint_dir):
+            print "Trying to force-restore checkpoint from inexistent directory", maybe_checkpoint_dir
```
Comment (Cwiiis) [inline util/automation.py]: s/inexistent/non-existent/

File: util/automation.py (hunk 1 above)
Comment (lissyx) [inline util/automation.py]: fixed!

File: util/automation.py (hunk 2 above)
Comment (lissyx) [inline util/automation.py]: fixed!

**Pull request merged**